### ISSUE TITLE
right shift count warning in winpr/libwinpr/file/generic.c for 32 bit builds

### ISSUE THREAD

**Issue by hendwolt** [FreeRDP/FreeRDP#3913]
This is a follow up for #3903, where this was the second of two issues.
After the first issue got fixed, this is still open:

[   60s] /home/abuild/rpmbuild/BUILD/FreeRDP-master/winpr/libwinpr/file/generic.c: In function 'FindNextFileA':
[   60s] /home/abuild/rpmbuild/BUILD/FreeRDP-master/winpr/libwinpr/file/generic.c:979:53: warning: right shift count >= width of type [-Wshift-count-overflow]
[   60s]     lpFindFileData->nFileSizeHigh = fileStat.st_size >> 32;
[   60s]                                                      ^~

Happens in 32 bit builds only.
[osTWbuild_log_20170411.zip](https://github.com/FreeRDP/FreeRDP/files/914319/osTWbuild_log_20170411.zip)

**Comment by hendwolt** [comment]
I investigated this a little further and think, it may be a fallout of an openSUSE bug (#1007123), where someone (I hope by accident) switched off the support for files larger than 2GB in some programs and libraries for 32bit.

TL;DR:
To change this, a compiler flag _FILE_OFFSET_BITS=64 needs to be set. 

On the other hand, I don't know, if there are any architectures out there with no large file support. If such architectures exists, this 32 bit shift should not be done unconditionally.

**Comment by hendwolt** [comment]
If I want to enable _FILE_OFFSET_BITS=64 for gcc (only), I use the attached patch
[cmake_64bit.patch.zip](https://github.com/FreeRDP/FreeRDP/files/962201/cmake_64bit.patch.zip)

**Comment by bmiklautz** [comment]
@hendwolt should be fixed with the current master if I'm not mistaken. Can you verify that?

**Comment by hendwolt** [comment]
Yes, it is fixed.
